libs/libvtrutil/src/vtr_thread_pool.h

```diff
@@ -0,0 +1,171 @@
+#pragma once
+
+/** 
+ * @file RouterThreadPool.h
+ * @brief A generic thread pool for parallel task execution
+ */
+
+#include <thread>
+#include <queue>
+#include <mutex>
+#include <condition_variable>
+#include <memory>
+#include <atomic>
+#include <functional>
+#include <cstddef>
+#include <vector>
+#include "vtr_log.h"
+#include "vtr_time.h"
+
+namespace vtr {
+
+class thread_pool {
+private:
```

Comment:
Comment every member variable with doxygen and the overall class.  I'd give an example of how to use the class in the overall comment.
This should also go into the vtr utility documentation as a new tab eventually.